### PR TITLE
Issue 1606: feat(testing): saving project json in db and validate 

### LINKED ISSUE #1606: Automated test part 3 - public projects
- Insert a public project into the DB (currently we only do fields, we need a way to insert a whole json file) - also put the date in the future so it appears at the top of the list
- Insert a private project and check it does not appear in the list

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1606

### Notes

A summary of what was achieved in this PR

- [ ] Saving public project json in db 
- [ ] Validate public project title on UI
- [ ] Saving private project json in db
- [ ] Validate private project title should not be visible on UI

### CODE REVIEW

File: web/test/data/public-project.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "type": "public",
+  "title": "Demo public project",
+  "description": "Demo public project description added",
+  "url": "https://www.pipelinedashboard.io"
+}
```
Comment (eddiejaoude) [inline web/test/data/public-project.json:6]: I was initially thinking of a complete project, but I think this is better, it tests that the application can handle minimum data. I will create new tickets for more details data like repositories etc

File: web/test/integration/project.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Feature:  Save the project json in db
```
Comment (eddiejaoude) [inline web/test/integration/project.feature]: This does not describe the epic/result we are expecting, I would I suggest something like `Display projects on the homepage`

File: web/test/integration/project.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Feature:  Save the project json in db 
+
+Scenario: Ensure the create public project
```
Comment (eddiejaoude) [inline web/test/integration/project.feature]: The test here is not the create, that is the step def, the test is the reading. I would suggest something like `List public project on the homepage`

File: web/test/plugins/index.js [hunk 4]
```diff
@@ -20,4 +20,12 @@ module.exports = (on, config) => {
     }
   });
 
+  on('task', {
+    'db:update:project': (params) => {
```
Comment (eddiejaoude) [inline web/test/plugins/index.js]: I suggest `db:project:save`

File: web/test/data/public-project.json (hunk 1 above)
Comment (webkhushboo) [reply web/test/data/public-project.json:6]: okay

File: web/test/integration/project.feature (hunk 2 above)
Comment (webkhushboo) [reply web/test/integration/project.feature]: okay will change the text

File: web/test/integration/project.feature (hunk 3 above)
Comment (webkhushboo) [reply web/test/integration/project.feature]: okay will update

File: web/test/plugins/index.js (hunk 4 above)
Comment (webkhushboo) [reply web/test/plugins/index.js]: okay will do

**Review by eddiejaoude — APPROVED** [review]

**Pull request merged**